rocAL/rocAL_pybind/amd/rocal/fn.py

```diff
@@ -29,219 +29,437 @@
+    exposure_image = b.Exposure(Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
     return (exposure_image)
 
-def fish_eye(*inputs, device=None):
+
+def fish_eye(*inputs, rocal_tensor_output_layout=types.NHWC, rocal_tensor_output_datatype=types.UINT8):
+    """
+    inputs - the input image passed to the augmentation
+    
+    rocal_tensor_output_layout (int, optional, default = types.NHWC) - tensor layout for the augmentation output
+
+    rocal_tensor_output_datatype (int, optional, default = types.UINT8) - tensor dtype for the augmentation output
+    """
     # pybind call arguments
-    kwargs_pybind = {"input_image0": inputs[0],"is_output": False}
-    fisheye_image = b.FishEye(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+    kwargs_pybind = {"input_image": inputs[0], "is_output": False,
+                     "rocal_tensor_output_layout": rocal_tensor_output_layout, "rocal_tensor_output_datatype": rocal_tensor_output_datatype}
+    fisheye_image = b.FishEye(Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
     return (fisheye_image)
 
-def fog(*inputs, fog=0.5, device=None):
+
+def fog(*inputs, fog=0.5, rocal_tensor_output_layout=types.NHWC, rocal_tensor_output_datatype=types.UINT8):
+    """
+    inputs - the input image passed to the augmentation
+    
+    fog (float, default = 0.5) - fog fill value used for the augmentation
+
+    rocal_tensor_output_layout (int, optional, default = types.NHWC) - tensor layout for the augmentation output
+
+    rocal_tensor_output_datatype (int, optional, default = types.UINT8) - tensor dtype for the augmentation output
+    """
+    fog = b.createFloatParameter(fog) if isinstance(fog, float) else fog
     # pybind call arguments
-    fog = b.CreateFloatParameter(fog) if isinstance(fog, float) else fog
-    kwargs_pybind = {"input_image0": inputs[0],
-                     "is_output": False, "fog_value": fog}
-    fog_image = b.Fog(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+    kwargs_pybind = {"input_image": inputs[0],
+                     "is_output": False, "fog_value": fog, "rocal_tensor_output_layout": rocal_tensor_output_layout, "rocal_tensor_output_datatype": rocal_tensor_output_datatype}
+    fog_image = b.Fog(Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
     return (fog_image)
 
-def brightness(*inputs, brightness=1.0, bytes_per_sample_hint=0, image_type=0,
-               preserve=False, seed=-1, device=None):
+
+def brightness(*inputs, alpha=None, beta=None, rocal_tensor_output_layout=types.NHWC, rocal_tensor_output_datatype=types.UINT8):
     """
-    brightness (float, optional, default = 1.0) –
+    inputs - the input image passed to the augmentation
+    
+    alpha (float, optional, default = None) - brightness multiplier. Values >= 0 are accepted. For example: 0 - black image, 1 - no change, 2 - increase brightness twice
```

Comment:
Are we sure there are no default params for brightness?

Reply:
Unless the user specifies a float value, this is how its been done in C++ so we have the done the same here